vsintegration/tests/UnitTests/HintServiceTests.fs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace VisualFSharp.UnitTests.Editor.Hints
+
+open NUnit.Framework
+open HintTestFramework
+
+module HintServiceTests =
+
+[<Test>]
+let ``Type hints are not shown in signature files`` () =
+    let fsiCode = """
+module Test
+
+val numbers: int[]
+"""
+    let fsCode = """
+module Test
+
+let numbers = [|42|]
+"""
+    let fsiDocument, _ = getFsiAndFsDocuments fsiCode fsCode
+
+    let result = getHints fsiDocument
+    
+    Assert.IsEmpty(result)
+
+[<Test>]
+let ``Type hint is shown for a let binding`` () =
+    let code = """
+type Song = { Artist: string; Title: string }
+
+let s = { Artist = "Moby"; Title = "Porcelain" }
+"""
+    let document = getFsDocument code
+    let expected = [{ Content = ": Song"; Location = (3, 6) }]
+
+    let actual = getHints document
+
+    Assert.AreEqual(expected, actual)
+
+[<Test>]
+let ``Type hint is not shown for a let binding when the type is manually specified`` () =
+    let code = """
+type Song = { Artist: string; Title: string }
+
+let s: Song = { Artist = "Moby"; Title = "Porcelain" }
+"""
+    let document = getFsDocument code
+    
+    let result = getHints document
+    
+    Assert.IsEmpty(result)
+
+[<Test>]
+let ``Type hint is shown for a parameter`` () =
+    let code = """
+type Song = { Artist: string; Title: string }
+
+let whoSings s = s.Artist
```

Comment:
The parameter "s" gets a type hint, and the whole "let whoSigns" does not get it?

If you do e.g. " let print = Console.Write ", i.e. let-bound function, does it produce any type hint or not?
(I am fine if this gets done in a later stage, but would be good to know and to have it covered by tests)

Reply:
That doesn't produce a hint yet. I added a test for that :)